src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/SqlParser.java

```diff
@@ -208,18 +209,34 @@ private void findOrderBy(MySqlSelectQueryBlock query, Select select) throws SqlP
 
     private void addOrderByToSelect(Select select, List<SQLSelectOrderByItem> items, String alias) throws SqlParseException {
         for (SQLSelectOrderByItem sqlSelectOrderByItem : items) {
-            SQLExpr expr = sqlSelectOrderByItem.getExpr();
-            Field f = FieldMaker.makeField(expr, null, null);
-            String orderByName = f.toString();
-
             if (sqlSelectOrderByItem.getType() == null) {
                 sqlSelectOrderByItem.setType(SQLOrderingSpecification.ASC);
             }
             String type = sqlSelectOrderByItem.getType().toString();
 
+            SQLExpr expr = sqlSelectOrderByItem.getExpr();
+            Field field = FieldMaker.makeField(expr, null, null);
+
+            String orderByName;
+            if (field instanceof MethodField && field.getName().equals("script")) {
```

Comment:
np: And to complement my suggestion with another point, this condition could be just replaced with `field.isScriptField()` .

Reply:
I moved that to separate method.